SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/sqlparser/CalciteCompiler.java

```diff
@@ -184,15 +184,21 @@ public CalciteCompiler(CompilerOptions options) {
         this.astRewriter = new RewriteDivision();
         this.options = options;
 
+        final boolean CASE_SENSITIVE = true;
+        final Casing UNQUOTED_CASING = Casing.UNCHANGED;
+        final Casing QUOTED_CASING = Casing.UNCHANGED;
         Properties connConfigProp = new Properties();
-        connConfigProp.put(CalciteConnectionProperty.CASE_SENSITIVE.camelName(), Boolean.TRUE.toString());
-        connConfigProp.put(CalciteConnectionProperty.UNQUOTED_CASING.camelName(), Casing.UNCHANGED.toString());
-        connConfigProp.put(CalciteConnectionProperty.QUOTED_CASING.camelName(), Casing.UNCHANGED.toString());
+        connConfigProp.put(CalciteConnectionProperty.CASE_SENSITIVE.camelName(), Boolean.toString(CASE_SENSITIVE));
+        connConfigProp.put(CalciteConnectionProperty.UNQUOTED_CASING.camelName(), UNQUOTED_CASING.toString());
+        connConfigProp.put(CalciteConnectionProperty.QUOTED_CASING.camelName(), QUOTED_CASING.toString());
         connConfigProp.put(CalciteConnectionProperty.CONFORMANCE.camelName(), SqlConformanceEnum.BABEL.toString());
         CalciteConnectionConfig connectionConfig = new CalciteConnectionConfigImpl(connConfigProp);
         SqlConformance conformance = connectionConfig.conformance();
         this.parserConfig = SqlParser.config()
                 .withLex(options.ioOptions.lexicalRules)
+                .withCaseSensitive(connectionConfig.caseSensitive())
```

Comment:
FYI I'm also happy if the JSON just gives back the original string the user typed (and not make it uppercase). 

AFAICT this change makes the SQL queries itself case-sensitive which may be strange to folks used to postgres, mysql, mssql etc.

I do understand the two things may be hard to separate though.

Reply:
What's the motivation here? Practically all databases people work with are case insensitive, so why are we trying to offer that?

Reply:
See the attached issue

Reply:
It seems like we are baking in one way of casing into the whole compiler. This should be a configuration option supplied from the commandline I think? Ideally, the compiler should internally have both the canonicalized and original names. The returned json schema can also have both names (or just one of them based on flags like "-js --preserve-original-casing"). The UX and manager shouldn't have to canonicalize by themselves when needed (e.g., when searching for a table).

Reply:
If something "should" happen, then someone should file an issue for that, and it will eventually happen.
Right now I am fixing issue 144.

Reply:
I'm suggesting that this fix include a configuration option as my review.

Reply:
Can you elaborate on what the flag(s) should be and what the expected behavior is?
I don't really understand how Calcite uses this stuff internally at this point.

Reply:
From the Calcite docs https://calcite.apache.org/docs/reference.html#identifiers, it looks like:
* unquoted identifiers are converted to upper case
* quoted identifiers retain their case.
* I think #144 is concerned with unquoted identifiers. If so, a configuration flag like "--unquoted-casing" to toggle between preserving the "upper-case" (the default of unquoted identifiers converted to upper case) or "case-sensitive" (unquoted ids retain case) makes sense.

To that end, I hope these three lines actually do regulate that behavior:

```
connConfigProp.put(CalciteConnectionProperty.CASE_SENSITIVE.camelName(), Boolean.toString(CASE_SENSITIVE));
connConfigProp.put(CalciteConnectionProperty.UNQUOTED_CASING.camelName(), UNQUOTED_CASING.toString());
connConfigProp.put(CalciteConnectionProperty.QUOTED_CASING.camelName(), QUOTED_CASING.toString());
```